File: leetcode/easy/typescript/moveZeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {
if (nums.length >= 1) {
      let readPointer = 0;
      for (let index = 0; index < nums.length; index++) {
        if (nums[readPointer] === 0) {
          nums.splice(readPointer, 1);
          nums.push(0);
        } else {
          readPointer++;
        }
      }
    }
};

File: leetcode/easy/cSharp/MoveZeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int left = 0;
        int right = 0;

        while (right < nums.Length)
        {
           
            if (nums[right] != 0)
            {
                int temp = nums[left]; nums[left] = nums[right]; nums[right] = temp;
            }
            if (nums[left] != 0) left++;
            right++;
        }
    }
}
